src/main/java/org/mule/extension/db/internal/domain/connection/mysql/MySqlConnectionParameters.java

```diff
@@ -115,8 +116,16 @@ public String getUser() {
   private void addMuleLoggerProperty(Map<String, String> connectionProperties) {
     if (connectionProperties != null) {
       try {
-        Thread.currentThread().getContextClassLoader().loadClass(MY_SQL_LOGGER);
-        connectionProperties.putIfAbsent(LOGGER_PROPERTY, MY_SQL_LOGGER);
+        Class<?> finalInterface = getAvailableInterface();
+        String arg = "MySql";
+        Enhancer e = new Enhancer();
+        e.setSuperclass(MuleMySqlLogger.class);
+
+        e.setInterfaces(new Class[] {finalInterface});
+        e.setCallback((MethodInterceptor) (o, method, objects, methodProxy) -> null);
```

Comment:
at which point does this delegate into the `MuleMySqlLogger` methods? Doesn't this simply returns null without invoking the super method? Do you have an actual test that verifies that the logging is happening?

Reply:
Changed callback with a NoOp instance which doesn't generate an intercepted method in the subclass generated from MuleMySqlLogger, thus the super method will be called directly instead.

Reply:
remains unchanged

Reply:
pushed